### DISCUSSION TITLE
How to add more unique buttons in app?

### DISCUSSION THREAD

**Discussion by Davidm16** [flet-dev/flet#1689] [Q&A] [answered]
### Question

Hi, I would like to know if it is possible to add more buttons in app, that each one is independent and that they can return unique values. This example adds 10 buttons but always gives the last value

### Code sample

```python
import flet as ft

def main(page: ft.Page):
    r = ft.Row(wrap=True, scroll="always", expand=True)
    page.add(r)
    
    def pruebas(e):
        print(e)

    for i in range(10):
        r.controls.append(ft.ElevatedButton(i, on_click=lambda _:pruebas(i)),)
    page.update()

ft.app(target=main)
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment, marked as answer]
You can do this:
```python
    for i in range(10):
        r.controls.append(ft.ElevatedButton(i, data=i, on_click=lambda e: pruebas(e.control.data)))
```

**Reply by Davidm16** [reply]
It works thanks!
